src/Documentation/resources/Best_Practices.md

```diff
@@ -0,0 +1,422 @@
+# Best Practices
+
+Orleans was designed in 2014 with the intention of removing low-level
+abstractions from actor model frameworks. Grains (actors stored in memory) are
+used to maintain the lifecycle among other features. Orleans is based on the
+actor model and fits some projects better that others.
+
+## Orleans should be considered when:
+
+-   Significant number (hundreds to millions) of loosely coupled entities
+
+    -   Example: Internet of Things (IoT) – individual devices turning on/off
+
+-   Entities are small enough to be single-threaded
+
+    -   Example: Determine if stock should be purchased based on current price
+
+-   Workload is interactive
+
+    -   Example: request-response, start/monitor/complete
+
+-   More than one server is expected or may be required\\
+
+    -   Orleans runs on a cluster which is expanded by adding servers to expand
+        the cluster
+
+-   Global coordination is not needed or on a smaller scale between a few
+    entities at a time
+
+    -   Orleans maintains its clusters and talks to other clusters. Each cluster
+        is able to communicate through tcp/ip
+
+-   *Various entities are used at different times*
+
+    -   This depends on the entities that are being used
+
+## Orleans is not the best fit when:
+
+-   Memory must be shared between entities
+
+    -   Each grain maintains its own states and should not be shared.
+
+-   A small number of large entities that may be multithreaded
+
+    -   A microservice may be a better option when supporting complex logic in a
+        single service
+
+-   Global coordination and/or consistency is needed
+
+    -   Orleans maintains the silos within its own framework
+
+-   *Operations that run for a long time*
+
+    -   Batch jobs, Single Instruction Multiple Data (SIMD) tasks
+
+    -   This depends on the need of the application and may be a fit for Orleans
+
+## Grains
+
+**Overview**:
+
+-   Actors are not objects, however they are similar
```

Comment:
As a general rule, we try to refrain from talking about actors. I think it is better to talk about grains instead.

I would also start with a positive statement. For example, "Grains resemble objects. However, they are distributed, virtual, and asynchronous."

Reply:
That's a much better way to put it!

Reply:
corrected